.github/workflows/cypress.yml

```diff
@@ -0,0 +1,87 @@
+name: Cypress Tests
+
+on:
+  pull_request:
+    types: [opened]
+  push:
+    branches:
+      - '**'
+
+jobs:
+  # TODO: remove once speed/batch size testing complete
+  # interval of 6 files:
+  #       specs: [ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21 ]
+  # interval of 5 files:
+  #       specs: [ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25 ]
+  # interval of 4 files:
+  #       specs: [ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31 ]
+  # interval of 3 files:
+  #       specs: [ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41 ]
+
+  e2e_tests:
+    
+    runs-on: ubuntu-latest
+
+    permissions:
+      actions: read
+
+    defaults:
+      run:
+        shell: bash -ileo pipefail {0}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        specs: [ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21 ]
+    
+    steps:
+      - name: Checkout Streamlit code
+        uses: actions/checkout@v3
+        with: 
+          persist-credentials: false
+          submodules: 'recursive'
+      - name: Set up Python 3.7.12
+        uses: actions/setup-python@v4
+        with:
+            python-version: "3.7.12"
+      - name: Setup virtual env
+        uses: ./.github/actions/make_init
+      - name: Run make develop
+        run: make develop
+      - name: Register Streamlit User & Mapbox Token
+        run: |
+          if [ ! -d $HOME/.streamlit ] ; then
+            mkdir $HOME/.streamlit
+          fi
+          MAPBOX_TOKEN=$(curl -sS https://data.streamlit.io/tokens.json | jq -r '.["mapbox-localhost"]') \
+          && echo '[mapbox]' >  ~/.streamlit/config.toml \
+          && echo 'token = "'$MAPBOX_TOKEN'"' >> ~/.streamlit/config.toml
```

Comment:
nit(s):

* Only L58 uses `MAPBOX_TOKEN`, so the echo in L57 doesn't have to be run in the same command and can be done in the line above
* the lines beginning with `&&` should be indented (2 spaces or 4 spaces.. I don't remember what bash convention is) to signify that they're continuations of L57